cns/ipampool/monitor.go

```diff
@@ -266,7 +267,7 @@ func (pm *Monitor) increasePoolSize(ctx context.Context, meta metaState, state i
 
 	if _, err := pm.nnccli.UpdateSpec(ctx, &tempNNCSpec); err != nil {
 		// caller will retry to update the CRD again
-		return err
+		return fmt.Errorf("[ipam-pool-monitor] Increasing pool size: UpdateCRDSpec failed %w", err)
```

Comment:
The other thing with the error strings is that I find it's best to just say what you were trying to do when the error was produced... since that's all the information you have at that exact point in the codebase. For example, something like:

```go
return errors.Wrap(err, "executing UpdateSpec")
```

Something _else_ can add the "increasing pool size:" bit, if that's indeed what it was trying to do upstream. The result is basically an inline stack trace of things you actually care about and can grep for: `[some-service] error reticulating splines: err: increasing pool size: executing UpdateSpec: invalid spec format`